core/src/main/java/com/google/cloud/sql/core/SslSocketFactory.java

```diff
@@ -141,13 +142,13 @@ public static synchronized SslSocketFactory getInstance() {
       }
 
       String userToken = System.getProperty(USER_TOKEN_PROPERTY_NAME);
-      Credential credential = credentialFactory.create();
+      if (userToken != null) {
+        applicationName = userToken;
```

Comment:
Can you do this inline where you call `setApplicationName()`, and then avoid updating a static variable?